packages/hebao_v1/contracts/modules/security/WhitelistModule.sol

```diff
@@ -0,0 +1,113 @@
+/*
+
+  Copyright 2017 Loopring Project Ltd (Loopring Foundation).
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+pragma solidity ^0.5.11;
+
+import "../../iface/Wallet.sol";
+
+import "../stores/WhitelistStore.sol";
+
+import "./SecurityModule.sol";
+
+
+/// @title WhitelistModule
+/// @dev Manages whitelisted addresses.
+contract WhitelistModule is SecurityModule
+{
+    WhitelistStore  public whitelistStore;
+    SecurityStore   public securityStore;
+
+    constructor(
+        SecurityStore  _securityStore,
+        WhitelistStore _whitelistStore
+        )
+        public
+        SecurityModule(_securityStore)
+    {
+        whitelistStore = _whitelistStore;
+    }
+
+    function addToWhitelist(
+        address wallet,
+        address addr
+        )
+        external
+        nonReentrant
+        onlyFromMetaTxOrWalletOwner(wallet)
+        onlyWhenWalletUnlocked(wallet)
+    {
+        whitelistStore.addToWhitelist(wallet, addr);
```

Comment:
I would think this would need a time delay or `numGuardiants + 1 / 2` signatures to be safe.

Reply:
I'll create an issue to apply a delay period.